platform/ios/src/MGLMapboxEvents.m

```diff
@@ -485,10 +500,15 @@ - (void)postEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)events {
             if (error) {
                 [strongSelf pushDebugEvent:MGLEventTypeLocalDebug withAttributes:@{MGLEventKeyLocalDebugDescription: @"Network error",
                                                                                         @"error": error}];
+                [[UIApplication sharedApplication] endBackgroundTask:_backgroundTaskIdentifier];
+                _backgroundTaskIdentifier = UIBackgroundTaskInvalid;
                 return;
```

Comment:
Thanks again. d648594 refactors as you described and squashes this all into a single commit on top of the release branch.